src/System.Management.Automation/engine/EventManager.cs

```diff
@@ -673,22 +664,10 @@ private void ProcessNewSubscriber(PSEventSubscriber subscriber, Object source, s
 
             if (_eventAssembly == null)
             {
-#if !CORECLR
-                // Define the assembly that will hold our event handlers
-                StackFrame callStack = new StackFrame(0, true);
-                debugMode = (callStack.GetFileName() != null);
```

Comment:
Note: `callstack.GetFileName()` return null if the corresponding `.pdb` file is not put beside the `.dll` file. So this is actually to see if we are currently debugging the `.dll`.